src/transformation/visitors/switch.ts

```diff
@@ -29,7 +31,20 @@ export const transformSwitchStatement: FunctionVisitor<ts.SwitchStatement> = (st
 
         const goto = lua.createGotoStatement(`${switchName}_case_${index}`);
         const conditionalGoto = lua.createIfStatement(condition, lua.createBlock([goto]));
-        statements.push(conditionalGoto);
+
+        resultIfStatements.push(conditionalGoto);
+    }
+
+    if (resultIfStatements.length > 0) {
```

Comment:
Please use `statements.reduce((previousIf, currentIf) => (previousIf.elseBlock = currentIf, currentIf));` instead if this entire if statement if it works

Reply:
Gotem